docs/developing_iree/e2e_benchmarking.md

```diff
@@ -0,0 +1,354 @@
+          │       # for TFLite's benchmark_model to use.
+          ├── traced_function_2
+          └── ...
+
+# Example for MatrixOpsStaticModule:
+/tmp/iree/modules/MatrixOpsStaticModule
+  ├── iree_llvmjit
+  │   ├── compiled.vmfb
+  │   └── traces
+  │       ├── basic_matmul
+  │       │   └── flagfile
+  │       ├── matmul_broadcast_singleton_dimension
+  │       │   └── flagfile
+  │       ├── matmul_lhs_batch
+  │       │   └── flagfile
+  │       └── matmul_rhs_batch
+  │           └── flagfile
+  ├── iree_vmla
+  │   ├── compiled.vmfb
+  │   └── traces  # ...same as iree_llvmjit/traces above.
+  ├── iree_vulkan
+  │   ├── compiled.vmfb
+  │   └── traces  # ...same as iree_llvmjit/traces above.
+  └── tflite
+      ├── basic_matmul.tflite
+      ├── matmul_broadcast_singleton_dimension.tflite
+      ├── matmul_lhs_batch.tflite
+      ├── matmul_rhs_batch.tflite
+      └── traces
+          ├── basic_matmul
+          │   └── graph_path
+          ├── matmul_broadcast_singleton_dimension
+          │   └── graph_path
+          ├── matmul_lhs_batch
+          │   └── graph_path
+          └── matmul_rhs_batch
+              └── graph_path
+```
+
+### Optional: Compile the Keras Applications Vision tests.
+
+The vision tests take a while to run, so we exclude them from the CI and
+wildcard expansion. They can be run by invoking the following test suite:
+
+```shell
+bazel test //integrations/tensorflow/e2e/keras:vision_external_tests
+```
+
+The previous command compiles `MobileNet`, `MobileNetV2` and `ResNet50` to run
+on `cifar10` and `imagenet` weights on all backends. The artifacts generated by
+this test suite are slightly different than those above in that they are
+organized by `/tmp/iree/modules/ModelName/Dataset/backends` instead of just by
+`/tmp/iree/modules/ModelName/backends`.
+
+## 2. Benchmarking IREE on desktop
+
+### 2.1 Build the `iree-benchmark-module`
+
+This step is optional, but allows running the benchmarks without running `bazel`
+at the same time.
+
+```shell
+bazel build -c opt //iree/tools:iree-benchmark-module
+```
+
+This creates `bazel-bin/iree/tools/iree-benchmark-module`. The rest of the guide
+will use this binary, but you could also use
+`bazel run iree/tools:iree-benchmark-module` in its place if your prefer.
+
+### 2.2 Benchmark the model on IREE
+
+The E2E tests generate a flagfile with all of the information that
+`iree-benchmark-module` needs to benchmark each trace. Namely it handles
+providing the following flags:
+
+| Flag              | Description                                      |
+|-------------------|--------------------------------------------------|
+| --input_file      | Path to the IREE compiled VM flatbuffer          |
+| --inputs          | A comma delimited string of input tensors        |
+| --driver          | The backend driver to use for the benchmark      |
+| --entry_function  | The method on the TensorFlow module to benchmark |
+
```

Comment:
Add a note that the flagfile references the abosolute paths so one might want to modify it if moving the artifacts around?

Reply:
I added a note that it's an absolute path, but kept the discussion about moving it in the android section so that there's a concrete example to go alongside it.